chemprop/cli/train.py

```diff
@@ -636,6 +639,83 @@ def build_splits(args, format_kwargs, featurization_kwargs):
     return train_data, val_data, test_data
 
 
+def summarize(args, dataset: _MolGraphDatasetMixin) -> tuple[list, list]:
+    if args.task_type in ["regression", "regression-mve", "regression-evidential"]:
+        if isinstance(dataset, MulticomponentDataset):
+            y = dataset.datasets[0].Y
+        else:
+            y = dataset.Y
+        y_mean = np.nanmean(y, axis=0)
+        print("YMEAN", y_mean)
+        y_std = np.nanstd(y, axis=0)
+        y_median = np.nanmedian(y, axis=0)
+        mean_dev_abs = np.abs(y - y_mean)
+        frac_1_sigma = np.sum((mean_dev_abs < y_std), axis=0) / len(y)
+        frac_2_sigma = np.sum((mean_dev_abs < 2 * y_std), axis=0) / len(y)
+
+        column_headers = ["Statistic"] + [f"Value {i}" for i in range(y.shape[1])]
+        table_rows = [
+            ["Num. datapoints"] + [f"{len(y)}" for i in range(y.shape[1])],
+            ["Mean"] + [f"{mean:0.2f}" for mean in y_mean],
```

Comment:
Limiting the printout to two decimals is a good idea to not get too large of numbers. Would it be possible to print 2 or 3 significant figures instead? Right now we get numbers like 0.03 and 156.95, when it could look better to print numbers like 0.0314 and "157.". I guess if a number gets small enough or big enough we would want to print it scientific notation, but for numbers around 0.01-100, I think scientific notation makes things harder to read and shouldn't be the default. This is just a nice to have so if you can't find a way to do it, I wouldn't worry about it.

Reply:
I added the following function to help address this...
```
def format_number(num):
    if 0.01 < np.abs(num) < 100:
        return f"{num:.2f}"
    else:
        return f"{num:.2e}"
 ```

Reply:
I think you're looking for the `g` [format specifier](https://docs.python.org/3/library/string.html#format-specification-mini-language):

> General format. For a given precision p >= 1, this rounds the number to p significant digits and then formats the result in either fixed-point format or in scientific notation, depending on its magnitude. A precision of 0 is treated as equivalent to a precision of 1.
>
> The precise rules are as follows: suppose that the result formatted with presentation type 'e' and precision p-1 would have exponent exp. Then, if m <= exp < p, where m is -4 for floats and -6 for [Decimals](https://docs.python.org/3/library/decimal.html#decimal.Decimal), the number is formatted with presentation type 'f' and precision p-1-exp. Otherwise, the number is formatted with presentation type 'e' and precision p-1. In both cases insignificant trailing zeros are removed from the significand, and the decimal point is also removed if there are no remaining digits following it, unless the '#' option is used.
>
> With no precision given, uses a precision of 6 significant digits for [float](https://docs.python.org/3/library/functions.html#float). For [Decimal](https://docs.python.org/3/library/decimal.html#decimal.Decimal), the coefficient of the result is formed from the coefficient digits of the value; scientific notation is used for values smaller than 1e-6 in absolute value and values where the place value of the least significant digit is larger than 1, and fixed-point notation is used otherwise.
> 
> Positive and negative infinity, positive and negative zero, and nans, are formatted as inf, -inf, 0, -0 and nan respectively, regardless of the precision.

Ex:

```python
>>> format_number = lambda x: f"{x:0.3g}"
>>> format_number(123.456)
'123'
>>> format_number(1234)
'1.23e+03'
>>> format_number(.0123)
'0.0123'
```